lib/dbutils.js

```diff
@@ -431,7 +425,50 @@ dbu.buildGetQuery = function(tableName, query, schema, includePreparedForDelete)
         }
         sql += condition + constructOrder(query, schema) + limit;
     }
-    return {sql: sql, params: params};
+    return sql;
+};
+
+function extractConditionParams(query, schema) {
+    var params = [];
+    var pred = query.attributes;
+    Object.keys(pred).forEach(function(predKey) {
+        var predObj = pred[predKey];
+        if (!predObj || predObj.constructor !== Object) {
+            params.push(schema.converters[schema.attributes[predKey]].write(predObj));
+            pred[predKey] = null;
+        } else {
+            var predKeys = Object.keys(predObj);
+            var predOp = predKeys[0];
+            var predArg = predObj[predOp];
+            if (predOp.toLowerCase() === 'between') {
+                params.push(schema.converters[schema.attributes[predKey]].write(predArg[0]));
+                params.push(schema.converters[schema.attributes[predKey]].write(predArg[1]));
+                predArg[0] = null;
+                predArg[1] = null;
+            } else {
+                params.push(schema.converters[schema.attributes[predKey]].write(predArg));
+                predObj[predOp] = null;
+            }
+        }
+    });
+    return params;
+}
+
+dbu.extractGetParams = function(query, schema, includePreparedForDelete) {
+    var params;
+    if (query.attributes) {
+        params = extractConditionParams(query, schema);
+    } else {
+        params = [];
+    }
+
+    // Also include check that _exist_until not expired
+    if (includePreparedForDelete) {
+        params.push(new Date().getTime());
+    } else {
+        query.includePreparedForDelete = false;
```

Comment:
It's a marker in a result of a `JSON.stringify()`. When we set it we get `"includePreparedForDelete":false there, when not set - we don't, so the resulting string (key for a cache) is different. And here monomorphism doesn't matter, as this object is always different.